### PR TITLE
Redesign MCQ choosing

### PR DESCRIPTION
**Pull request by remo5000**
### Features
- Only MCQ option are on the MCQChooser component
- MCQ Questions are differentiated based on the presence of a `solution` (and hints):
  - Buttons have intent when the question has a solution
  - Hints are displayed as notifications when options are chosen
  - For ungraded questions, there is no intent on the buttons and there is no notification displayed.

### CODE REVIEW

File: src/components/workspace/MCQChooser.tsx [hunk 1]
```diff
@@ -53,16 +47,49 @@ class MCQChooser extends React.PureComponent<IMCQChooserProps, State> {
...
+  /**
+   * Handles the logic for what intent an MCQ option should show up as.
+   * This is dependent on the presence of an actual solution (for ungraded assessments),
+   * the current selection, and whether the selected option is active.
+   *
+   * @param currentOption the current button key, corresponding to a choice ID
+   * @param chosenOption the mcq option that is chosen in the state, i.e what should show up as "selected"
+   * @param solution the solution to the mcq, if any
+   */
+  private getButtonIntent = (
```
Comment (remo5000) [inline src/components/workspace/MCQChooser.tsx]: If possible could you review this with a bit more care? It looks ok to me, but feels like there might be come edge cases I haven't accounted for @ningyuansg

File: src/mocks/assessmentAPI.ts [hunk 2]
```diff
@@ -179,7 +179,35 @@ export const mockAssessmentQuestions: Array<IProgrammingQuestion | IMCQQuestion>
...
+    solution: 2
+  },
+  {
+    answer: 3,
+    content:
+      'This is the 4rth question. Oddly enough, it is a graded MCQ question that uses the curves library!',
+    choices: [
+      {
+        content: 'A',
+        hint: 'hint A'
```
Comment (remo5000) [inline src/mocks/assessmentAPI.ts]: Remove hints for ungraded question. Inconsequential, but semantics matter.

File: src/styles/_workspace.scss [hunk 3]
```diff
@@ -82,19 +82,15 @@ $code-color-error: #ff4444;
...
-    .mcq-option {
-      // use of !important to override a 'not' selector by BluePrint
-      background-color: $cadet-color-3 !important;
-      padding: 20px;
-    }
-    .mcq-task-parent {
-      margin-bottom: 20px;
-      .pt-card {
-        background-color: $cadet-color-3;
+      .mcq-options-parent {
```
Comment (remo5000) [inline src/styles/_workspace.scss]: Nested some classes here, let me know if it's alright

File: src/components/workspace/MCQChooser.tsx [hunk 4]
```diff
@@ -23,26 +24,19 @@ class MCQChooser extends React.PureComponent<IMCQChooserProps, State> {
...
-            <div className="mcq-task-parent row center-xs ">
-              <Card className="mcq-task col-xs-12" elevation={2}>
-                <Text className="Text"> {this.props.mcq.content} </Text>
-              </Card>
-            </div>
-            <div className="row mcq-options-parent center-xs">{options}</div>
-          </div>
+      <div className="MCQChooser row">
+        <Card className="mcq-content-parent col-xs-12 middle-xs">
+          <div className="row mcq-options-parent between-xs">{options}</div>
```
Comment (ningyuansg) [inline src/components/workspace/MCQChooser.tsx]: Put in button group to avoid weird gaps between the options.

**Pull request merged**